cmfive.php

```diff
@@ -171,7 +237,37 @@ function stepOneYieldsWeb()
     return false;
 }
 
-function installCoreLibraries()
+function refreshComposerAvailability()
+{
+    // in case Composer has pulled new packages
+    // after the autoloader already went up from Web
+    // (when core is installed, web is used for module configs dependencies)
+    // (so autoloader needs flushing when the modules have finally come in)
+    $allIncluded = get_declared_classes();
+    foreach ($allIncluded as $key => $class) {
+        //because our ComposerAutoloader class has a suffixed long-hash:
+        if (strpos($class, "ComposerAutoloaderInit") !== false) {
+            $composerLoader = $class;
+            // OK we found it, so apply this stolen code from:
+            // composer\vendor\composer\autoload_real.php
+            $map = require 'composer/vendor/composer/autoload_psr4.php';
+            foreach ($map as $namespace => $path) {
+                // a minimal response, as Psr4 gets us
+                // at least the db classes needed for web
+                $composerLoader::getLoader()->setPsr4($namespace, $path);
+            }
+            echo "\nFlicked Composer switch: " . $class . "\n";
+        }
+    }
+}
+
+function cmdinstallCoreLibraries($pCount, $parameters = [])
+{
+    $parameters = array_slice($parameters, 2);
+    installCoreLibraries($parameters[0]);
```

Comment:
This should really check if ```$parameters``` is not empty. Since you're also just slicing the array and getting the first element this is also akin to accessing an unsliced ```$parameters``` as position 2. I.e. replace all of this with:
```php
installCoreLibraries(count($parameters) >= 3 ? $parameters[2] : <default value or error state>);
```

Reply:
is 'argc, argv' C-type CLI params, so have changed it to check with $pCount as trusted.